### DISCUSSION TITLE
Pikka - free to download / I need your help to improve Pikka.

### DISCUSSION THREAD

**Discussion by rsayfutdinov** [rsayfutdinov/Pikka#3] [Announcements]
Hello everyone,

I'm Ruslan - an indie developer behind Pikka.
Honestly, Pikka was not updated for a long time for many reasons.
I have a long list of issues to be fixed and improvements to be made. But first I would like to ask for your help. I need your feedback. 
I would like to know what you are missing in Pikka, how you use it, what can I do to make Pikka better. So don't hesitate to start a new discussion here.

I've decided to make the next version free to download for everyone but will include "PRO" features, that need to be purchased for the same price. All users who already have purchased Pikka do not need to unlock PRO features again. If you already purchased Pikka and see the "Upgrade to PRO" window in the app, **please contact me**.

Thank you,
Ruslan

**Comment by unwittinglyrad** [comment]
Hi Ruslan, 
App updated for me overnight and am now stuck with the "Upgrade to Pro" window. I purchased the App back in 2018. Please advise how to fix, thanks.

**Reply by rsayfutdinov** [reply]
Hello,

Sorry that it didn't work for you immediately.

There is two options:

1) I'm already working on an update, that hopefully fix your issues. It will be available on the app store within next 4-7 days. So you need to wait a little.

2) I drop the price to 0.99 for a while, you purchase it again. And I refund the purchase via Pay Pal. I know it’s not perfect solution but, should work very quick. Let me know if you want to go with this one.

Thanks for understanding ,
Ruslan.

**Reply by rsayfutdinov** [reply]
Hey unwittinglyrad,
Reach me at support@1sense.app
I will send you a promo code to use in the app. So you do not need to wait for an update.
Ruslan.

**Reply by unwittinglyrad** [reply]
> Hey unwittinglyrad,
> Reach me at [support@1sense.app](mailto:support@1sense.app)
> I will send you a promo code to use in the app. So you do not need to wait for an update.
> Ruslan.

Thanks Ruslan, email sent.

**Comment by jeffwalker99** [comment]
Hey, Ruslan.

If I pick a colour that is 221,84,79 (#ef4748 in hex), Pikka changes that to "Pale Red #dd544f". 

I can't figure out why it is doing this. Is it a bug?

**Reply by rsayfutdinov** [reply]
Hey Jeff,

#dd544f seems to be a right hex value for rgb(221,84,79).

Google gives the same result:
<img width="873" alt="Screenshot 2021-06-21 at 16 22 46" src="https://user-images.githubusercontent.com/12187524/122778060-ed289600-d2ac-11eb-978b-2a36f8743397.png">

**Reply by jeffwalker99** [reply]
Oh. I didn't check that.

In that case Pikka is picking up the incorrect colour.

On the web page, the background colour of the container is set to #ef4748, but Pikka is picking it up as #dd544f....

![image](https://user-images.githubusercontent.com/4571369/122780644-f1ea4b80-d2a6-11eb-8761-d6222e07b870.png)

Please try it for yourself, here: https://ninjaforms.com/

Try grabbing the red colour in the header.

Weird, eh?

**Reply by rsayfutdinov** [reply]
Is that Safari?

I've tried it and get the #ef4748 value. Look at the screen recording attached. 
Do you pick colors from the second monitor or on the MacBook screen?

https://user-images.githubusercontent.com/12187524/122782168-c8ceb880-d2b0-11eb-8a01-3b7c0b6a85d2.mov

It seems to be a color space conversion issue.

**Reply by jeffwalker99** [reply]
I'm using an iMac. I'm using Brave mostly, but I've checked and it's exactly the same in every browser.

Why am I seeing RGB values when I am picking a colour, and in your screen recording you are seeing hex values?

![image](https://user-images.githubusercontent.com/4571369/122783502-88b80780-d2a9-11eb-8d91-8cd462909045.png)

I've been trying this on quite a few websites and I'm having the same problem everywhere.

This is weirding me out! 😃

**Reply by jeffwalker99** [reply]
When I use a Chrome extension to pick colours (I've tried several) I always get the correct values.

**Reply by rsayfutdinov** [reply]
> I'm using an iMac. I'm using Brave mostly, but I've checked and it's exactly the same in every browser.
> 
> 
> 
> Why am I seeing RGB values when I am picking a colour, and in your screen recording you are seeing hex values?
> 
> 
> 
> ![image](https://user-images.githubusercontent.com/4571369/122783502-88b80780-d2a9-11eb-8d91-8cd462909045.png)
> 
> 
> 
> I've been trying this on quite a few websites and I'm having the same problem everywhere.
> 
> 
> 
> This is weirding me out! 😃 

You are seeing RGB and I get hex values because I'm using the newest version, that is not on AppStore yet😁

I will investigate the issue and come back to you.

Regards,
Ruslan

**Reply by jeffwalker99** [reply]
Thanks!

**Reply by rsayfutdinov** [reply]
Hey Jeff,

Issue solved! Thank you for pointing that out.
The update will be available this week at the App Store.

**Reply by jeffwalker99** [reply]
Awesome!

**Reply by rsayfutdinov** [reply]
The update is on the AppStore now. Please let me know if it fixes your issue.

Regards,
Ruslan.

**Reply by jeffwalker99** [reply]
Unfortunately, no, the issue is still there...

![image](https://user-images.githubusercontent.com/4571369/123086815-c6d53880-d41b-11eb-9678-eedd7b13ae76.png)

So where the background colour is set to #ef4748 Pikka is seeing #f04748. This is different to before, but it is still incorrect.

Another example from the same website, the yellow background of the button is #f2f061 but Pikka is seeing #f3ef61.

Here's an example from the BT Sport website (https://www.bt.com/sport)...

![image](https://user-images.githubusercontent.com/4571369/123088158-4d3e4a00-d41d-11eb-8aad-814e14546cb1.png)

One last example, this from the Kobo website (https://www.kobo.com/)...

![image](https://user-images.githubusercontent.com/4571369/123088721-000ea800-d41e-11eb-8d7c-0cbee388e3bf.png)

Yet on the BBC Sport website (https://www.bbc.co.uk/sport) it gets it right...

![image](https://user-images.githubusercontent.com/4571369/123089430-d73ae280-d41e-11eb-99a7-0f5af84daab8.png)

Weird. eh?

Especially weird because if I use a browser colour picker extension, it always sees the correct colour.

**Reply by rsayfutdinov** [reply]
Hmm.
Would you mind sharing your display color profile? You can find it under macOS Settings->Displays->Colour->Display profile.

**Reply by jeffwalker99** [reply]
No worries. Here you go...

![image](https://user-images.githubusercontent.com/4571369/123099774-67325980-d42a-11eb-8227-a8fb577787ca.png)

**Reply by rsayfutdinov** [reply]
> Especially weird because if I use a browser colour picker extension, it always sees the correct colour.

I think browser extensions just copy the color values from the CSS. On the other hand native apps like Pikka need to grab pixels from the screen, convert it into appropriate color space. Therefore they are not always accurate.

Nevertheless, it is really strange, cause I get the right values from Pikka now for all examples you gave.

Is that iMac 5K that you use?

**Reply by jeffwalker99** [reply]
Yes, it is a Retina 5K display...

![image](https://user-images.githubusercontent.com/4571369/123101535-1459a180-d42c-11eb-8ea0-1177cc322457.png)

I think it must be a colour profile thing. If I select the "Apple RGB" profile, Pikka sees more colours correctly, and the colours that are incorrect are very close to being correct. Going through the list of profiles, the one that results in Pikka seeing most colours correctly is the "sRGBIEC61966-2.1" profile.

Which colour profile are you using?

**Reply by rsayfutdinov** [reply]
I've tested it on the MacBook PRO screen, MacBook Air M1 screen, on LG 4K and 5K Monitors with different color profiles.
Only with Adobe RGB(1998) Profile, I get the wrong values.

I will add custom color profiles to Pikka soon, so one could choose the profile manually. Hopefully, it will fix the issue then.

**Reply by jeffwalker99** [reply]
That sounds like a great idea!

**Comment by chadcribbins** [comment]
Hey Ruslan, im still trying to figure out how to get a refund?
